uitools/cpp/Esri/ArcGISRuntime/Toolkit/SearchViewController.cpp

```diff
@@ -618,6 +641,26 @@ namespace Toolkit {
                 {
                   if (results.empty())
                     return;
+                  auto connection = std::make_shared<QMetaObject::Connection>();
+                  //connect either to a scene or a map event changedviewpoint
+                  if (auto mapView = qobject_cast<MapViewToolkit*>(m_geoView))
+                  {
+                    *connection = connect(mapView, &MapViewToolkit::viewpointChanged, this, [mapView, connection, this]()
+                                          {
+                                            disconnect(*connection);
+                                            auto extent = mapView->currentViewpoint(ViewpointType::BoundingGeometry).targetGeometry().extent();
+                                            m_lastSearchArea = extent;
+                                          });
+                  }
+                  else if (auto sceneView = qobject_cast<SceneViewToolkit*>(m_geoView))
+                  {
+                    *connection = connect(sceneView, &SceneViewToolkit::viewpointChanged, this, [sceneView, connection, this]()
+                                          {
+                                            disconnect(*connection);
+                                            auto extent = sceneView->currentViewpoint(ViewpointType::BoundingGeometry).targetGeometry().extent();
+                                            m_lastSearchArea = extent;
+                                          });
+                  }
```

Comment:
🔧  Make these `singleShot`

Reply:
I already made them in my other PR. I think you are right in making them `singleshot` here!